Documentation/gettingstarted/kind.rst

```diff
@@ -0,0 +1,224 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _gs_kind:
+
+**************************
+Getting Started Using Kind
+**************************
+
+This guide uses `kind <https://kind.sigs.k8s.io/>`_ to demonstrate deployment
+and operation of Cilium in a multi-node Kubernetes cluster.
+
+Kind requires docker to be installed and running.
+
+Install Docker, kubectl, & kind
+===============================
+
+1. Install ``docker`` stable as described in: `Install Docker Engine <https://docs.docker.com/engine/install/>`_
+
+2. Install ``kubectl`` version >= v1.14.0 as described in the `Kubernetes Docs <https://kubernetes.io/docs/tasks/tools/install-kubectl/>`_.
+
+3. Install ``kind`` >= v0.7.0 per kind documentation: `Installation and Usage <https://kind.sigs.k8s.io/#installation-and-usage>`_
+
+Kind Configuration
+==================
+
+Kind doesn't use flags for configuration. Instead it uses YAML configuration that is very similar to Kubernetes.
+
+Create a ``kind-config.yaml`` file based on the following template. The template will create 3 node + 1 apiserver
+cluster with the latest version of kubernetes from when the kind release was created.
+
+.. code:: yaml
+
+    kind: Cluster
+    apiVersion: kind.x-k8s.io/v1alpha4
+    nodes:
+    - role: control-plane
+    - role: worker
+    - role: worker
+    - role: worker
+    networking:
+      disableDefaultCNI: true
+
+To change the version of kubernetes being run,  ``image`` has to be defined for each node. See
+the `Node Configration <https://kind.sigs.k8s.io/docs/user/configuration/#nodes>`_ documentation.
+
+Start Kind
+==========
+
+Pass the ``kind-config.yaml`` you created with the ``--config`` flag of kind.
+
+.. code:: bash
+
+    kind create cluster --config=kind-config.yaml
+
+This will add a ``kind-kind`` context to ``KUBECONFIG`` or if unset ``${HOME}/.kube/config``
+
+.. code:: bash
+
+    kubectl cluster-info --context kind-kind
+
+.. _kind_install_cilium:
+
+Install Cilium
+==============
+
+.. include:: k8s-install-download-release.rst
+
+.. note::
+
+    As Kind is running nodes as containers in Docker, they're sharing your host machines' kernel.
+    So we have to disable :ref:`host-services` otherwise kind clusters will clobber each other
```

Comment:
Should we pass `--set global.hostServices.enabled=false` below? I don't see it in the example helm command.

Reply:
Actually this should only be necessary for the multi-cluster case right? Unless we could also define the service IP range in the kind config, in which case we can get rid of this restriction entirely?

Reply:
(And now I read the multicluster, they do exactly this. Do we need a note at all about host services then?)

Reply:
I'll add an explicit host services disabled below and move the note to a troubleshooting section.

What happens is, if cilium is installed into kind without disabling host services, it can break all kind clusters current or new. Even those not using cilium. It's easy enough to fix (restart or run the cleanup script).

Reply:
This context (what happens and how to fix) might be suitable for a "Troubleshooting" section at the bottom of the page.

Reply:
aaaand that's exactly what you did. Nice :-)